central/reprocessor/reprocessor.go

```diff
@@ -460,9 +462,28 @@ func (l *loopImpl) reprocessNodes() {
 }
 
 func (l *loopImpl) reprocessWatchedImage(name string) bool {
-	img, err := imageEnricher.EnrichImageByName(emptyCtx, l.imageEnricher, imageEnricher.EnrichmentContext{
+	enrichmentCtx := imageEnricher.EnrichmentContext{
 		FetchOpt: imageEnricher.IgnoreExistingImages,
-	}, name)
+	}
+
+	ctx := emptyCtx
+	if features.DelegateWatchedImageReprocessing.Enabled() {
+		// Delegated scanning performs an access check, an empty context will
+		// trigger a hard panic. If an image watch was successfully created
+		// we can assume at that time the user had access, so we create a
+		// context that has access to delegate the scan.
```

Comment:
nit: when I first read this I thought these were two separate thoughts

```suggestion
		// Delegated scanning performs an access check, and an empty context will
		// trigger a hard panic. If an image watch was successfully created
		// we can assume at that time the user had access, so we create a
		// context that has access to delegate the scan.
```

Reply:
another thought: while you tried to explain the access stuff by calling out "we can assume that time the user had access"; however this generally runs in the "backend" scope and isn't tied to a user, so I would either omit it or be more explicit (e.g. something along the lines of `a user with previous access rights configured the watched image reprocessing, thus we share the same access with the context, this context isn't shared with any client`).

Reply:
https://github.com/stackrox/stackrox/pull/11121/commits/6d1869c6592f8a16157135e8d18dac7a39f368f0 - I removed the comments due to the confusion.